paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/PrimaryKeyLookupTable.java

```diff
@@ -99,25 +99,35 @@ public void refresh(Iterator<InternalRow> incremental) throws IOException {
         Predicate predicate = projectedPredicate();
         while (incremental.hasNext()) {
             InternalRow row = incremental.next();
-            primaryKeyRow.replaceRow(row);
-            if (userDefinedSeqComparator != null) {
-                InternalRow previous = tableState.get(primaryKeyRow);
-                if (previous != null && userDefinedSeqComparator.compare(previous, row) > 0) {
-                    continue;
+            if (refreshAsync) {
+                synchronized (lock) {
```

Comment:
Is it possible to achieve thread safety in `RocksDBState`, it looks like only the inconsistency may occur when updating the `cache`.

Reply:
Currently there will be `primaryKeyRow`, `RocksDBSetState` and `RocksDBListState` in `PrimaryKeyLookupTable`,  `SecondaryIndexLookupTable` and `NoPrimaryKeyLookupTable`, they are not thread safe. I think we can keep the lock in `get` method and optimize it when we find that it has become a bottleneck. What do you think?